### PR TITLE
jewel: osd: also check the exsistence of clone obc for "CEPH_SNAPDIR" requests

### PR DESCRIPTION
**Pull request by liewegas**
http://tracker.ceph.com/issues/17445

### CODE REVIEW

File: src/osd/ReplicatedPG.cc [hunk 1]
```diff
@@ -2054,7 +2054,7 @@ void ReplicatedPG::do_op(OpRequestRef& op)
 	}
 
 	ObjectContextRef sobc = get_object_context(clone_oid, false);
-	if (!sobc) {
+	if (!sobc || !sobc->obs.exists) {
```
Comment (liu-chunmei) [inline src/osd/ReplicatedPG.cc]: in what operation sobc exists, but sobc->obs.exists is false?

Comment (xxhdx1985126) [reply src/osd/ReplicatedPG.cc]: This happens after a clone object evicted from the cache tier

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-10-02_07:55:29-rados-wip-kefu-testing-2017-10-02-1347-jewel-distro-basic-mira/

**Comment by xxhdx1985126** [comment]
Um, hi, can anyone review this PR, please:-)

**Comment by xxhdx1985126** [comment]
@smithfarm I'm working on it:-)

**Comment by smithfarm** [comment]
This PR passed a rados integration run at http://tracker.ceph.com/issues/21742#note-14 with two bugs reproducing: 

* http://tracker.ceph.com/issues/23006
* http://tracker.ceph.com/issues/23007

@xxhdx1985126 @liu-chunmei @jdurgin @tchaikov Do you think it's ready to merge?

**Pull request merged**